gitrevise/merge.py

```diff
@@ -175,6 +178,67 @@ def merge_blobs(
     repo = current.repo
 
     tmpdir = repo.get_tempdir()
+
+    annotated_labels = (
+        f"{path} (new parent): {labels[0]}",
+        f"{path} (old parent): {labels[1]}",
+        f"{path} (current): {labels[2]}",
+    )
+    (is_clean_merge, merged) = merge_files(
+        annotated_labels, current, base, other, tmpdir
+    )
+
+    if is_clean_merge:
+        # No conflicts.
+        return Blob(repo, merged)
+
+    # At this point, we know that there are merge conflicts to resolve.
+    # Prompt to try and trigger manual resolution.
+    print(f"Conflict applying '{labels[2]}'")
+    print(f"  Path: '{path}'")
+
+    preimage = merged
+    (normalized_preimage, conflict_id, merged_blob) = replay_recorded_resolution(
+        repo, tmpdir, preimage
+    )
+    if merged_blob is not None:
+        return merged_blob
+
+    if input("  Edit conflicted file? (Y/n) ").lower() == "n":
+        raise MergeConflict("user aborted")  # pylint: disable=W0707
+
+    # Open the editor on the conflicted file. We ensure the relative path
+    # matches the path of the original file for a better editor experience.
+    conflicts = tmpdir / "conflict" / path.relative_to("/")
+    conflicts.parent.mkdir(parents=True, exist_ok=True)
+    conflicts.write_bytes(preimage)
+    merged = edit_file(repo, conflicts)
+
+    # Print warnings if the merge looks like it may have failed.
+    if merged == preimage:
+        print("(note) conflicted file is unchanged")
+
+    if b"<<<<<<<" in merged or b"=======" in merged or b">>>>>>>" in merged:
+        print("(note) conflict markers found in the merged file")
+
+    # Was the merge successful?
+    if input("  Merge successful? (y/N) ").lower() != "y":
+        raise MergeConflict("user aborted")  # pylint: disable=W0707
+
+    record_resolution(repo, conflict_id, normalized_preimage, merged)
+
+    return Blob(current.repo, merged)
+
+
+def merge_files(
+    labels: Tuple[str, str, str],
+    current: Blob,
+    base: Optional[Blob],
+    other: Blob,
```

Comment:
nit: I think these three arguments would make more sense as `bytes` arguments, as we're currently being forced to create `Blob` objects from bytes in `replay_recorded_resolution` just to pass them to this function.